.github/workflows/publish.yaml

```diff
@@ -1,19 +1,17 @@
+# Basic flow where we do all generic tests:
```

Comment:
Couldn't we merge the `publish.yaml` and `publish-dry-run.yaml` files to only keep one `publish.yaml` file and decrease redundancy?

We could:
- Use a [list of event](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-using-a-list-of-events).
- [Condition a step](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsif) with an evaluation of `github.event_name`. (Or even just [condition the command arguments](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepswithargs)).

Reply:
Done in https://github.com/tchataigner/mock-crate/pull/4/commits/4306137d2ef02b2ce6359e83b847714d9c970fce